### PR TITLE
Cleanliness is next to godliness

### PR DESCRIPTION
**Pull request by ZenGround0**
This PR removes the last of the ugly store test setup in favor of the new chain builder.

While trying to close out #2222 I found myself making laborious fixes to this code and realizing that it was all scheduled to be erased.  It seemed like as good a time as any to do this.

#2128 was called out in the chain store test comments as tracking cleanup.  I don't think we should close that with this merging because there is some unresolved and healthy discussion there.

### CODE REVIEW

File: chain/store_test.go [hunk 1]
```diff
@@ -406,94 +182,90 @@ func TestGetTipSetState(t *testing.T) {
...
+	builder := chain.NewBuilder(t, address.Undef)
+	genTS := builder.NewGenesis()
+	r := repo.NewInMemoryRepo()
+	cs := newChainStore(r, genTS.At(0).Cid())
+
+	// Construct test chain data
+	link1 := builder.AppendOn(genTS, 2)
+	link2 := builder.AppendOn(link1, 3)
+	link3 := builder.AppendOn(link2, 1)
+	link4 := builder.BuildOn(link3, 2, func(bb *chain.BlockBuilder, i int) { bb.IncHeight(2) })
```
Comment (ZenGround0) [inline chain/store_test.go]: I could abstract out the frequently repeated lines from 187 to 196 if reviewers prefer.

File: chain/store_test.go [hunk 2]
```diff
@@ -9,273 +9,37 @@ import (
...
 	"github.com/filecoin-project/go-filecoin/state"
-	th "github.com/filecoin-project/go-filecoin/testhelpers"
 	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
-// This setup stuff is a horrendous legacy holdover from when these store tests shared a
-// common monolithic setup with the syncer tests. There's no good reason for it to still exist.
-// https://github.com/filecoin-project/go-filecoin/issues/2128
-type SyncerTestParams struct {
```
Comment (frrist) [inline chain/store_test.go]: :heart: :1st_place_medal: :fireworks: YOU ROCK

Comment (shannonwells) [reply chain/store_test.go]: ❤️ 🎧 🎸

File: chain/store_test.go (hunk 1 above)
Comment (anorth) [reply chain/store_test.go]: No need IMO, but you could pull a lot of these simpler tests into `t.Run()` sharing the same builder/links and chain.

**Pull request merged**